### DISCUSSION TITLE
Issues with out of the box EKS cluster following the docs

### DISCUSSION THREAD

**Discussion by p5150j** [milvus-io/milvus#27782] [Q&A and General discussion]
Following the docs here
https://milvus.io/docs/eks.md

```
apiVersion: eksctl.io/v1alpha5
kind: ClusterConfig
metadata:
  name: my-eks-milvus-cluster
  region: eu-central-1
  version: "1.23"

nodeGroups:
  - name: ng-1-workers
    labels: { role: workers }
    instanceType: m5.4xlarge
    desiredCapacity: 2
    volumeSize: 80
    iam:
      withAddonPolicies:
        ebs: true

addons:
- name: aws-ebs-csi-driver
  version: v1.13.0-eksbuild.1 # optional


```
All went well, no drama... now trying to test this... locally we have:

```
pip show pymilvus

Name: pymilvus
Version: 2.3.1
Summary: Python Sdk for Milvus
Home-page: 
Author: 
Author-email: Milvus Team <milvus-team@[zilliz.com](http://zilliz.com/)>
```

Available collections: (these are small `Hello World` collections the team pushed)

```
demo
nhat_test
test_l2
```

And we have a simple script to connect and load

```
from pymilvus import connections, Collection

# Connect to Milvus
host = "......eu-central-1.elb.amazonaws.com"
port = 19530

# Connect to Milvus server
connections.connect(host=host, port=port)
print("Successfully connected to the Milvus server!")

# Load the 'test_l2' collection
collection_name = "test_l2"
try:
    collection = Collection(name=collection_name)
    collection.load()
    print(f"Loading collection {collection_name}...")

    # Check loading status
    if collection.is_loaded:
        print(f"Collection {collection_name} loaded successfully.")
    else:
        print(f"Failed to load collection {collection_name}.")
except Exception as e:
    print(f"An error occurred: {e}")
```

### But when I run this I get 

```
Successfully connected to the Milvus server!

.......
..................(wait 4 mins)
RPC error: [get_loading_progress], <MilvusException: (code=1, message=collection 444467791732931035 has not been loaded to memory or load failed)>, <Time:{'RPC start': '2023-10-12 16:35:39.374995', 'RPC error': '2023-10-12 16:35:39.524652'}>
RPC error: [wait_for_loading_collection], <MilvusException: (code=1, message=collection 444467791732931035 has not been loaded to memory or load failed)>, <Time:{'RPC start': '2023-10-12 16:25:38.995519', 'RPC error': '2023-10-12 16:35:39.527289'}>
RPC error: [load_collection], <MilvusException: (code=1, message=collection 444467791732931035 has not been loaded to memory or load failed)>, <Time:{'RPC start': '2023-10-12 16:25:38.822769', 'RPC error': '2023-10-12 16:35:39.527506'}>
An error occurred: <MilvusException: (code=1, message=collection 444467791732931035 has not been loaded to memory or load failed)>
````

# Milvus Loading Issue Diagnostic Report

**Objective:**  
Diagnose the issue where Milvus returns an error stating that a collection has not been loaded into memory or failed to load.

### 1. **Review of System Resources**
   - Instance Type: m5.4xlarge
   - Memory Available: 64 GiB
   - CPU Cores: 16

   **Action Taken:**
   Checked the memory and CPU utilization of the nodes running Milvus using the `kubectl top nodes` and `kubectl top pods` commands.

We have conducted a thorough analysis of the system, focusing on the memory utilization and allocation within the Kubernetes pods running on AWS EC2 instances of type m5.4xlarge. The m5.4xlarge instances are equipped with 64 GiB of memory and 16 vCPUs, ensuring a robust environment for memory-intensive tasks.

Detailed Findings
The memory consumption of each pod was monitored and logged using the kubectl top pods command, yielding real-time data on the resource utilization of each pod. The pod with the highest memory consumption, milvus-querynode-684cbfb5b7-z7vfn, was utilizing 854 MiB of memory, a fraction of the total available resources. Other pods exhibited even lower memory usage, affirming the ample availability of memory resources.

Here is a breakdown of the memory usage of some of the key pods:

- milvus-querynode-684cbfb5b7-z7vfn: 854Mi
- milvus-pulsar-broker-0: 2919Mi
- milvus-pulsar-bookie-0: 2606Mi
- ... (others have less memory usage)

   **Findings:**
   The memory utilization was within normal limits, indicating that the issue is not due to insufficient memory.

### 2. **Review of Milvus Configuration**
   - Milvus Version: v2.3.1 (confirmed as the latest version)

   **Action Taken:**
   Examined the Milvus configuration via ConfigMap in Kubernetes to ensure it is configured correctly for the existing system resources.

   **Findings:**
   The configuration appeared to be in order, with no apparent misconfigurations that could lead to the loading issue.

### 3. **Review of Milvus Version and Update**
   - Current Version: v2.3.1
   - Latest Available Version: v2.3.1 (at the time of diagnosis)

   **Action Taken:**
   Checked the currently deployed Milvus version and compared it with the latest available version on GitHub to determine if an update is necessary.

   **Findings:**
   The deployed Milvus version is up to date. No update is required at this time.

**Comment by yhmo** [comment]
Some questions:
- How many rows in the collection "test_l2"? 
- Dimension of each vector is?
- What kind of index was assigned to the vector field?

**Reply by p5150j** [reply]
The collection 'odin_test' has 314 vectors.
Each vector in 'odin_test' has a dimension of 1536.

The collection 'odin_beta' has 4149 vectors.
Each vector in 'odin_beta' has a dimension of 1536.

The collection 'odin_text_beta' has 15 vectors.
Each vector in 'odin_text_beta' has a dimension of 1536.

The collection 'demo' has 0 vectors.
Each vector in 'demo' has a dimension of 128.

The collection 'nhat_test' has 0 vectors.
Each vector in 'nhat_test' has a dimension of 128.

The collection 'test_l2' has 975 vectors.
Each vector in 'test_l2' has a dimension of 1536.

The collection 'odin_beta_images' has 117 vectors.
Each vector in 'odin_beta_images' has a dimension of 1536
